### PR TITLE
OCC-205: Add name parts to Address

### LINKED ISSUE #386: make class Address partial (OCC-205)
Hi,

need additonal fields in shipping and billing address.
Would it be possible to make the "Address" class a partial class to let me extend the class?

And how could AddressFormatter be overwritten to take care of additional fields in the class extension?

Thanks
Dieter

[Jira issue](https://lombiq.atlassian.net/browse/OCC-205)

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OCC-205](https://lombiq.atlassian.net/browse/OCC-205)
Fixes #386

### CODE REVIEW

File: src/Modules/OrchardCore.Commerce.ContentFields/Drivers/AddressFieldDisplayDriver.cs [hunk 1]
```diff
@@ -69,41 +82,44 @@ public class AddressFieldDisplayDriver : ContentFieldDisplayDriver<AddressField>
...
 
-        if (!updated ||
-            IsRequiredFieldEmpty(viewModel.Address.Name, nameof(viewModel.Address.Name)) ||
-            IsRequiredFieldEmpty(viewModel.Address.StreetAddress1, nameof(viewModel.Address.StreetAddress1)) ||
-            IsRequiredFieldEmpty(viewModel.Address.City, nameof(viewModel.Address.City)))
+        foreach (var addressUpdater in _addressUpdaters)
         {
-            return await EditAsync(field, context);
+            await addressUpdater.UpdateAsync(address);
         }
```
[the remark this answers is not in the document]
Comment (sarahelsaig) [reply src/Modules/OrchardCore.Commerce.ContentFields/Drivers/AddressFieldDisplayDriver.cs:105]: Sure, but why? Doesn't seem to improve readability in this case.

File: src/Modules/OrchardCore.Commerce.ContentFields/Manifest.cs [hunk 2]
```diff
@@ -8,3 +9,27 @@
...
+
+[assembly: Feature(
+    Id = ContentFields,
+    Name = "Orchard Core Commerce - Content Fields",
+    Category = "Commerce",
+    Description = "Commerce-specific content fields for Orchard Core.",
+    Dependencies = new[]
+    {
+        "OrchardCore.Commerce.ContentFields",
+    }
```
[the remark this answers is not in the document]
Comment (sarahelsaig) [reply src/Modules/OrchardCore.Commerce.ContentFields/Manifest.cs:21]: I think it's a mistake and I wanted to write `OrchardCore.ContentFields`.

File: src/Modules/OrchardCore.Commerce.ContentFields/Drivers/AddressFieldDisplayDriver.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (barthamark) [reply src/Modules/OrchardCore.Commerce.ContentFields/Drivers/AddressFieldDisplayDriver.cs]: When _does it_ improve readability then?

[the remark this answers is not in the document]
Comment (sarahelsaig) [reply src/Modules/OrchardCore.Commerce.ContentFields/Drivers/AddressFieldDisplayDriver.cs]: In arrow functions.

**Pull request merged**